File: LeetCode/summary-ranges.c
#include <stdio.h>

char **summaryRanges(int *nums, int numsSize, int *returnSize)
{
  char **res = malloc(sizeof(char *) * numsSize);

  int start = 0;
  *returnSize = 0;

  for (int i = 0; i < numsSize; i++)
  {
    while (i < numsSize - 1 && nums[i] + 1 == nums[i + 1])
      i++;

    int idx = *returnSize;
    res[idx] = calloc(25, sizeof(char));

    if (start == i)
      sprintf(res[idx], "%d", nums[i]);
    else
      sprintf(res[idx], "%d->%d", nums[start], nums[i]);

    start = i + 1;
    (*returnSize)++;
  }

  return res;
}

File: LeetCode/summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
  let start = 0;
  let end = 0;
  const res = [];
  for (let i = 1; i < nums.length; i++) {
    if (nums[end] + 1 === nums[i]) end++;
    else {
      if (start === end) res.push(`${nums[start]}`);
      else res.push(`${nums[start]}->${nums[end]}`);
      start = end = i;
    }
  }
  if (nums.length)
    res.push(start === end ? `${nums[start]}` : `${nums[start]}->${nums[end]}`);
  return res;
};
